cloud/services/compute/network.go

```diff
@@ -31,7 +32,9 @@ func (s *Service) ReconcileNetwork() error {
 	// Create Network
 	spec := s.getNetworkSpec()
 	network, err := s.networks.Get(s.scope.Project(), spec.Name).Do()
+	autoCreateCloudNat := false
 	if gcperrors.IsNotFound(err) {
+		autoCreateCloudNat = true
```

Comment:
This is a behavior change. If it is a concern, one option is we introduce another flag in networkspec to indicate whether we should auto create cloudNAT, with that flag default to false for now.

Reply:
That is true, however there has not been an official release for v0.3.0 yet, so behavioral changes may be fine. I will defer the decision to the community, though since I'm not sure how folks are currently using cluster-api-provider-gcp today and how the change might affect them.